### ISSUE TITLE
Error Unhandled Exception: type 'Future<dynamic>' is not a subtype of type 'FutureOr<GoogleSignInAccount>'

### ISSUE THREAD

**Issue by jeanJavaMan** [flutter/flutter#67244]
Hello guys, I apologize for my english, because I'm using google translator.

When adding to my package: firebase_messaging he requests in one of the steps that I add to my gradle the following excerpt:

`dependencies {
   // Example existing classpath
   classpath 'com.android.tools.build:gradle:3.5.3'
   // Add the google services classpath
   classpath 'com.google.gms: google-services: 4.3.2'
}`

when adding to the gradle and trying to use the google_sign_in package, I get the error:

**`Unhandled Exception: type Future <dynamic> is not a subtype of type FutureOr <GoogleSignInAccount>`**

find out that I just add `'com.google.gms: google-services: 4.3.2'` to the gradle that this happens, when I don't use it, login with google works normally.

part of the code I use to login:
`GoogleSignIn _googleSing = GoogleSignIn();
  var googleresult = await _googleSing.signIn().catchError(
        (error) => FlushbarHelper.createError(
                title: 'Erro ao logar',
                message: "Ocorreu um erro ao entrar em sua conta Google")
            .show(context),
      );`

could someone tell me why when adding to gradle: classpath **'com.google.gms: google-services: 4.3.2'** login with google stops working? And how to solve it.

**Comment by MrUnfunny** [comment]
IMO, this issue can be resolved by specifying type for googleresult. Try changing ```var googleresult = await _googleSing.signIn().catchError( (error) => FlushbarHelper.createError( title: 'Erro ao logar', message: "Ocorreu um erro ao entrar em sua conta Google") .show(context), );``` 
to
 ```GoogleSignInAccount googleresult = await _googleSing.signIn().catchError( (error) => FlushbarHelper.createError( title: 'Erro ao logar', message: "Ocorreu um erro ao entrar em sua conta Google") .show(context), );```

**Comment by jeanJavaMan** [comment]
> IMO, this issue can be resolved by specifying type for googleresult. Try changing `var googleresult = await _googleSing.signIn().catchError( (error) => FlushbarHelper.createError( title: 'Erro ao logar', message: "Ocorreu um erro ao entrar em sua conta Google") .show(context), );`
> to
> `GoogleSignInAccount googleresult = await _googleSing.signIn().catchError( (error) => FlushbarHelper.createError( title: 'Erro ao logar', message: "Ocorreu um erro ao entrar em sua conta Google") .show(context), );`

Hi @MrUnfunny 
I did as you said, but the error continues!

`GoogleSignIn _googleSing = GoogleSignIn();`
`  GoogleSignInAccount googleresult = await _googleSing.signIn().catchError(`
        `(error) => FlushbarHelper.createError(`
                `title: 'Erro ao logar',`
                `message: "Ocorreu um erro ao entrar em sua conta Google")`
            `.show(context),`
`      );`

Error:
`E/flutter (30919): [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: type 'Future<dynamic>' is not a subtype of type 'FutureOr<GoogleSignInAccount>'`
`E/flutter (30919): #0      _FutureListener.handleError  (dart:async/future_impl.dart:160:20))`
`E/flutter (30919): #1      Future._propagateToListeners.handleError  (dart:async/future_impl.dart:708:47))`
`E/flutter (30919): #2      Future._propagateToListeners  (dart:async/future_impl.dart:729:24))`
`E/flutter (30919): #3      Future._completeError  (dart:async/future_impl.dart:537:5))`
`E/flutter (30919): #4      _AsyncAwaitCompleter.completeError  (dart:async-patch/async_patch.dart:47:15))`
`E/flutter (30919): #5      GoogleSignIn._callMethod (package:google_sign_in/google_sign_in.dart)`
`E/flutter (30919): #6      _asyncErrorWrapperHelper.errorCallback  (dart:async-patch/async_patch.dart:91:64))`

Thanks for your answer

**Comment by jeanJavaMan** [comment]
Guys I want to apologize, but the error was just configuration.
I was using the credentials in Google Developer Console, when using 'com.google.gms: google-services: 4.3.2' it ended up giving a conflict with firebase, so the error was badly configured.
I resolved it as follows:
I deleted the credentials from the Google Developer Console, and just configured it on the Firebase Console. That done the problem was solved.

Here also helped me to solve this:
[https://stackoverflow.com/questions/54557479/flutter-and-google-sign-in-plugin-platformexceptionsign-in-failed-com-google#answer-54696963](url)
